r-lib-error-1.0.md

```diff
@@ -0,0 +1,64 @@
+
+# Specification of r-lib errors, version 1.0.0
+
+Note: this is an experimental draft.
+
+Note: version 1.0.0 corresponds to the `rlang_error` class, as
+implemented in rlang 0.4.0.
+
+## Classes
+
+An r-lib error object always has at least classes `rlib_error`, `error`
+and `condition`. It may have other classes.
+
+## Members
+
+An `rlib_error` object has the following members:
+
+* `message`: `character(1)`, optional, the error message. If this is
```

Comment:
Optional = `NULL` if not supplied? Or not present?

Reply:
Optional is not supplied. rlang currently sets `parent = NULL`, if there is no parent, so I think not present and `NULL` should be equivalent. I'll define this.